pkg/ident/ident.go

```diff
@@ -46,36 +46,36 @@ func marshalType(h hash.Hash, addressType AddressType) {
 
 func MarshalBytes(h hash.Hash, v []byte) {
 	marshalType(h, AddressTypeBytes)
-	MarshalInt64(h, int64(len(v)))
```

Comment:
Commenting here but it's about all the int -> uint changes in  `ident.go`. 
I don't see any test coverage for this pkg, and don't feel comfortable with this change - Also I'm not familiar with this code. 

Why are you changing this in the same PR?

Reply:
I wanted to fix the overflow warning and make the code clearer. However I had to keep the nolint comment so reverting